--- a/src/types/fetch.rs	2006-07-24 13:21:28
+++ b/src/types/fetch.rs	2026-09-21 21:12:35
@@ -259,6 +259,24 @@
                 .iter()
                 .filter_map(|av| match av {
                     AttributeValue::GmailMsgId(id) => Some(id),
+                    _ => None,
+                })
+                .next()
+        } else {
+            unreachable!()
+        }
+    }
+
+    /// Extract the `X-GM-THRID` of a `FETCH` response
+    ///
+    /// See [Access to the Gmail thread ID: X-GM-THRID](https://developers.google.com/workspace/gmail/imap/imap-extensions#access_to_the_gmail_thread_id_x-gm-thrid)
+    /// for details.
+    pub fn gmail_thr_id(&self) -> Option<&u64> {
+        if let Response::Fetch(_, attrs) = self.response.parsed() {
+            attrs
+                .iter()
+                .filter_map(|av| match av {
+                    AttributeValue::GmailThrId(id) => Some(id),
                     _ => None,
                 })
                 .next()
